execution_workflows/DaPars2/Dapars2_snakemake/workflow/scripts/results_to_bed.py

```diff
@@ -0,0 +1,182 @@
+             G---A   T   C   ...
+    BED    └─0   1 └─2   3   ...
+
+    If + strand Then the poly(A) site (loci_end) becomes
+    GFF      1 ┌─2─┐ 3   4   ...
+             G   A   T   C   ...
+    BED      0 └─1 └─2   3   ...
+
+    if - strand then poly(A) site (loci_start) becomes
+    GFF    ┌─1─┐ 2   3   4   ...
+             G   A   T   C   ...
+    BED    └─0 └─1   2   3   ...
+
+    i.e. if going from 1-based (GFF) --> BED (one_based = True)
+    + strand: start = loci_end - 1, end = loci_end
+    - strand: start = loci_start - 1, end = loci_start
+
+    if going from 0-based (BED) --> BED (one_based = False)
+    + strand: start = loci_end - 1, end = loci_end (end coordinates in BED are exclusive)
+    - strand: start = loci_start, end = loci_start + 1 (start coords in BED are inclusive)
+    '''
+
+    #First pull out start and end coords from 'Loci' column
+    #formatted like chr6:119923969-119926687
+
+    df[["loci_start", "loci_end"]] = (df[coord_col]
+                                      .str.split(":", expand = True) # df with cols of chr6 & 119923969-119926687
+                                      .loc[:,1] # 119923969-119926687 i.e. 2nd column
+                                      .str.split("-", expand = True) #df with cols 119923969 & 119926687
+                                      .astype(int)
+                                     )
+
+    # Select 3-prime most coordinate for each Tx from loci_start & end depending on strand
+    df = df.assign(three_p = lambda x: np.where(x[strand_col] == "+",
+                                                x["loci_end"],
+                                                x["loci_start"]
+                                               )
+                  )
+
+    # define BED start and end coords from 3-prime most coordinate
+    if one_based:
+        # GFF -> BED
+        df = df.assign(**{start_outcol: lambda row: row["three_p"] - 1,
+                          end_outcol: lambda row: row["three_p"]
+                         }
+                      )
+
+    else:
+        # BED of region to BED of single nucleotide
+        df = df.assign(**{start_outcol: lambda x: np.where(x[strand_col] == "+",
+                                                           x["three_p"] - 1,
+                                                           x["three_p"]
+                                                          ),
+                          end_outcol: lambda x: np.where(x[strand_col] == "+",
+                                                         x["three_p"],
+                                                         x["three_p"] + 1 # - strand: 3'most = loci_start, which is inclusive in BED
+                                                        )
+                         }
+                      )
+
+    return df.drop(["loci_start","loci_end","three_p"], axis = 1)
+
+
+def polya_site_id(df,
+                  cols_to_combine = ["transcript_id", "gene_symbol"],
```

Comment:
Probably the ID for a polyasite could be simpler, like chr:start:strand.... but as we don't need the IDs for the summary workflows (or do we?) it doesn't matter at all.

Reply:
Back during the hackathon I asked this and was told it doesn't matter what the ID is (the thread is in the `#tool-dapars2` Slack channel). You're right it probably could be simpler though
![image](https://user-images.githubusercontent.com/49978382/135267433-370ba56c-a2ba-4221-96d7-23e0f757acfb.png)